historicalData/historicalDataRepository.ts

```diff
@@ -0,0 +1,156 @@
+import type { Logger } from '@aws-lambda-powertools/logger'
+import {
+	QueryCommand,
+	type TimestreamQueryClient,
+} from '@aws-sdk/client-timestream-query'
+import {
+	Context,
+	HistoricalDataRequest,
+	HistoricalDataResponse,
+	chartProto,
+} from '@hello.nrfcloud.com/proto/hello'
+import { parseResult } from '@nordicsemiconductor/timestream-helpers'
+import { Type, type Static } from '@sinclair/typebox'
+import { Value } from '@sinclair/typebox/value'
+import { groupBy } from 'lodash-es'
+import { getQueryStatement } from './queryGenerator.js'
+
+export type HistoricalRequest = Omit<
+	Static<typeof HistoricalDataRequest>,
+	'data'
+>
+type HistoricalResponse = Static<typeof HistoricalDataResponse>
+
+export const transformTimestreamData = (
+	data: Record<string, unknown>[],
+	mapKeys: { fromKey: string; toKey: string }[],
+): Record<string, unknown>[] => {
+	const transformedData = []
+
+	// Added ts to mapKeys as default
+	mapKeys.push({ fromKey: 'time', toKey: 'ts' })
+
+	const groupedData = groupBy(data, (d) => d.time)
+	for (const item in groupedData) {
+		const data = groupedData[item]
+		const tranformedRecord = data?.reduce<Record<string, unknown>>(
+			(result, record) => {
+				result = {
+					...mapKeys.reduce<Record<string, unknown>>(
+						(resultMapKeys, mapKey) => {
+							if (mapKey.fromKey in record) {
+								if (mapKey.fromKey === 'time') {
+									resultMapKeys[mapKey.toKey] = (
+										record[mapKey.fromKey] as Date
+									).getTime()
+								} else {
+									resultMapKeys[mapKey.toKey] = record[mapKey.fromKey]
+								}
+							}
+							return resultMapKeys
+						},
+						{},
+					),
+					...result,
+				}
+				return result
+			},
+			{},
+		)
+		if (tranformedRecord !== undefined) transformedData.push(tranformedRecord)
+	}
+
+	return transformedData
+}
+
+export const historicalDataRepository = ({
+	timestream,
+	historicalDataTableInfo,
+	log,
+}: {
+	timestream: TimestreamQueryClient
+	historicalDataTableInfo: string
+	log?: Logger
+}): {
+	getHistoricalData: ({
+		deviceId,
+		model,
+		request,
+	}: {
+		deviceId: string
+		model: string
+		request: HistoricalRequest
+	}) => Promise<HistoricalResponse[]>
+} => {
+	const [historicalDataDatabaseName, historicalDataTableName] =
+		historicalDataTableInfo.split('|')
+	if (
+		historicalDataDatabaseName === undefined ||
+		historicalDataTableName === undefined
+	) {
+		throw new Error(
+			`Historical data table info must be in format "databaseName|tableName"`,
+		)
+	}
```

Comment:
Move parsing of the table info outside and have the API require table and database name, so it can be reused in other lambdas, and the function here does not need to throw an error.